x-pack/plugins/cloud/public/plugin.ts

```diff
@@ -82,19 +88,47 @@ export class CloudPlugin implements Plugin<CloudSetup> {
   public start(coreStart: CoreStart, { security }: CloudStartDependencies) {
     const { deployment_url: deploymentUrl, base_url: baseUrl } = this.config;
     coreStart.chrome.setHelpSupportUrl(ELASTIC_SUPPORT_LINK);
-    if (baseUrl && deploymentUrl) {
-      coreStart.chrome.setCustomNavLink({
-        title: i18n.translate('xpack.cloud.deploymentLinkLabel', {
-          defaultMessage: 'Manage this deployment',
-        }),
-        euiIconType: 'arrowLeft',
-        href: getFullCloudUrl(baseUrl, deploymentUrl),
-      });
-    }
 
-    if (security && this.isCloudEnabled) {
-      const userMenuLinks = createUserMenuLinks(this.config);
-      security.navControlService.addUserMenuLinks(userMenuLinks);
-    }
+    const setLinks = (authorized: boolean) => {
+      if (!authorized) return;
+
+      if (baseUrl && deploymentUrl) {
+        coreStart.chrome.setCustomNavLink({
+          title: i18n.translate('xpack.cloud.deploymentLinkLabel', {
+            defaultMessage: 'Manage this deployment',
+          }),
+          euiIconType: 'arrowLeft',
+          href: getFullCloudUrl(baseUrl, deploymentUrl),
+        });
+      }
+
+      if (security && this.isCloudEnabled) {
+        const userMenuLinks = createUserMenuLinks(this.config);
+        security!.navControlService.addUserMenuLinks(userMenuLinks);
```

Comment:
__optional nit:__ it seems we check that `security` exists in the `if` above (feel free to ignore, doesn't justify 1h+ CI run)
```suggestion
        security.navControlService.addUserMenuLinks(userMenuLinks);
```

Reply:
Bah, I needed this in an earlier refactoring, and forgot about it. I'll make this change, because I hate stray `!` (What is the carbon footprint of a CI run these days? 🌎 🔥)